### PR TITLE
Remove deprecation notice for index handling in vibe.web.rest

### PR DESCRIPTION
**Pull request by Geod24**
This had been deprecated in fe6a5de7433140660b14af28e9540b22c198c1c7 and so this message is there since July 2013 (and not relevant anymore). Time to remove it ?

### CODE REVIEW

**Comment by s-ludwig** [comment]
Instead of removing just the deprecation message, the whole `index` special case should be removed. For that we'd need something like a `disable_index_special_case = false` parameter for `extractHTTPMethodAndName`, because unfortunately the `vibe.web.web` module still needs to support it (although it was never supposed to be supported there in the first place).

**Comment by Geod24** [comment]
Since there was no other code, I though it was already removed. I'll take a look later this week then.

**Comment by Geod24** [comment]
Updated (and added small cosmetic changes).

File: source/vibe/web/rest.d [hunk 1]
```diff
@@ -32,22 +32,22 @@ import std.typetuple : anySatisfy, Filter;
...
+		$(TR $(TD add)    $(TD POST))
+		$(TR $(TD create) $(TD POST))
+		$(TR $(TD post)   $(TD POST))
+	)
 
 	If a method has its first parameter named 'id', it will be mapped to ':id/method' and
-    'id' is expected to be part of the URL instead of a JSON request. Parameters with default
-    values will be optional in the corresponding JSON request.
+        'id' is expected to be part of the URL instead of a JSON request. Parameters with default
+        values will be optional in the corresponding JSON request.
```
Comment (s-ludwig) [inline source/vibe/web/rest.d]: These should be tabs instead of spaces. I'll fix that after merge.

**Comment by s-ludwig** [comment]
There still seems to be a missing false template parameter at rest.d line 813 according to Travis.

**Comment by s-ludwig** [comment]
Nevermind. Your fix came in just in the second I commented.

**Comment by Geod24** [comment]
Yup, and that means the unittest don't cover that line. Reminds me that I never checked the coverage on Vibe.
It should be fixed now (and tabs are in).

**Comment by Geod24** [comment]
Green :)

**Pull request merged**